### PR TITLE
Issue 4561: Using S3 Rest Style Config URI for Extended S3 client construction

### LINKED ISSUE #4561: Use ECS Config URL for client configs
**Problem description**
Since Emc ECS Client 2.2.4, a mechanism exists that allows users to specify the complete ECS S3 client configuration in a single string by using a Config URI.
https://github.com/EMCECS/ecs-object-client-java/wiki/Config-URI-format
Pravega should take the advantage of flexibility provided by the config URI, to reduce the future code changes due to config item changes. 

**Problem location**
config.properties
ExtendedS3StorageConfig
ExtendedS3StorageFactory
ExtendecdS3Storage
init_tier2.sh

**Suggestions for an improvement**
Combine and consolidate all the config items (except Bucket and Prefix) into a single uri.

### PR DESCRIPTION
**Pull request by kevinhan88**
Signed-off-by: kevinhan88 <kevinhan88@gmail.com>

**Change log description**  
Started using s3 rest style configUri for majority of extendeds3 storage client config.

**Purpose of the change**  
Closes #4561 

**What the code does**  
(Detailed description of the code changes)
config.properties now contains configUri, bucket and prefix for extendedS3 storage.
ExtendedS3StorageConfig takes configUri and construct s3Config from it.
ExtendedS3StorageFactory handles configUri now.
ExtendedS3StorageConfigTest is added.
ExtendedS3StorageTest is updated to work with the new configUri
ExtendedS3IntegrationTest is updated to work with the new configUri

**How to verify it**  
(Optional: steps to verify that the changes are effective)
No functional changes. All tests (unit, integration, system and longevity) should pass.
The new configUri should start taking effect.

### CODE REVIEW

File: config/config.properties [hunk 1]
```diff
@@ -427,24 +427,25 @@ hdfs.hdfsUrl=localhost:9000
 
 ##region Extended S3 settings
 
-# URL where the extended S3 cluster is accessible at.
+# URI to specify the complete extended S3 client configurations (excludes bucket and prefix), in the format of
+# <protocol>://<host>[:<port>][/path][?<param-key>=<param-value>][&<param-key>=<param-value>][...]
+# Refer https://github.com/EMCECS/ecs-object-client-java/wiki/Config-URI-format for more details.
```
Comment (andreipaduroiu) [inline config/config.properties]: "Refer to"

File: config/config.properties [hunk 2]
```diff
@@ -427,24 +427,25 @@ hdfs.hdfsUrl=localhost:9000
...
+#   url: where the extended S3 cluster is accessible at, e.g. http://localhost:9020
+#   identity: the access key to access the extended S3 cluster, e.g. identity=user
+#   secretKey: the secret key to access the extended S3 cluster, e.g. secretKey=password
+extendeds3.configUri=http://localhost:9020?identity=user&secretKey=password
 
 # Shared extended S3 bucket where the data is stored.
 # This value must be the same for all Pravega SegmentStore instances in this cluster.
 # extendeds3.bucket=
 
+# Prefix in extended S3 cluster where all Pravega-related data for this cluster is following.
```
Comment (medvedevigorek) [inline config/config.properties]: Need to update the comment as it's a bit incorrect. A prefix in this case is a prefix that will be added to every object created by this instance of segment store in a configured S3 bucket. Using prefix allows sharing the same bucket between different Pravega clusters and also differentiate objects within a bucket created by Pravega from other objects. 

Also please mention that the prefix is optional and can be left empty.

File: config/config.properties (hunk 1 above)
Comment (kevinhan88) [reply config/config.properties]: fixed

File: config/config.properties (hunk 2 above)
Comment (kevinhan88) [reply config/config.properties]: fixed

**Pull request merged**